### PR TITLE
Add option to split by dups or using TreeKO algorithm

### PR DESCRIPTION
**Pull request by anilthanki**

### CODE REVIEW

File: tools/ete/ete_genetree_splitter.xml [hunk 1]
```diff
@@ -72,6 +77,7 @@ python '$__tool_directory__/ete_genetree_splitter.py'
         <test>
             <param name="genetreeFile" ftype="nhx" value="genetree.nhx" />
             <param name="gene_node" value="0"/>
+            <param name="splitter" value"dups"/>
```
Comment (nsoranzo) [inline tools/ete/ete_genetree_splitter.xml]:
```suggestion
            <param name="splitter" value="dups"/>
```

File: tools/ete/ete_genetree_splitter.xml [hunk 2]
```diff
@@ -83,6 +89,7 @@ python '$__tool_directory__/ete_genetree_splitter.py'
...
```
Comment (nsoranzo) [inline tools/ete/ete_genetree_splitter.xml]:
```suggestion
            <param name="splitter" value="dups"/>
```

File: tools/ete/ete_genetree_splitter.py [hunk 3]
```diff
@@ -47,11 +48,22 @@ def main():
...
-        with open(outfile, 'w') as f:
-            f.write(node.write(format=options.output_format))
+    if options.split == "dups":
+        # splits tree by duplication events which returns the list of all subtrees resulting from splitting current tree by its duplication nodes.
+        for cluster_id, node in enumerate(genetree.split_by_dups(), 1):
+            outfile = str(cluster_id) + '_genetree.nhx'
+            with open(outfile, 'w') as f:
+                f.write(node.write(format=options.output_format))
+    elif options.split == "treeko":
+        # splits tree by using TreeKO algorithms.
```
Comment (nsoranzo) [inline tools/ete/ete_genetree_splitter.py]:
```suggestion
        # splits tree using the TreeKO algorithm
```

**Comment by nsoranzo** [comment]
You need to change the tool version to `@VERSION@+galaxy1`

File: tools/ete/ete_genetree_splitter.xml [hunk 4]
```diff
@@ -25,6 +26,10 @@ python '$__tool_directory__/ete_genetree_splitter.py'
             <option value="0" selected="true">gene_species</option>
             <option value="1">species_gene</option>
         </param>
+        <param name="splitter" type="select" label="GeneTree splitting algorithm">
+            <option value="dups">Split by Duplication</option>
+            <option value="treeko" selected="true">Split using TreeKO algorithm</option>
```
Comment (nsoranzo) [inline tools/ete/ete_genetree_splitter.xml]: Should this be the new default or do we want to keep the old behaviour?

Comment (anilthanki) [reply tools/ete/ete_genetree_splitter.xml]: Either is fine

**Comment by anilthanki** [comment]
You mean **version="@VERSION@+galaxy1"** ?

That turns into  (Galaxy Version 3.1.1+galaxy1)

**Comment by nsoranzo** [comment]
Yes:
```xml
<tool id="ete_genetree_splitter" name="ETE GeneTree splitter" version="@VERSION@+galaxy1">
```

See https://galaxy-iuc-standards.readthedocs.io/en/latest/best_practices/tool_xml.html#tool-versions

File: tools/ete/ete_genetree_splitter.xml (hunk 4 above)
Comment (nsoranzo) [reply tools/ete/ete_genetree_splitter.xml]: I'd keep the old behaviour as default then:
```suggestion
            <option value="treeko">Split using TreeKO algorithm</option>
```

**Comment by nsoranzo** [comment]
Can you also add a test for the new feature, please?

**Comment by anilthanki** [comment]
Can I?... Yes, 

Do I want to.? No :smile: 

Give me some time

**Pull request merged**